src/inputs/buttons/toggle-button/PToggleButton.vue

```diff
@@ -1,82 +1,93 @@
-          :class="{'disabled': disabled}"
+    <label class="p-toggle-button"
+           :class="{'disabled': props.disabled}"
     >
-        <toggle-button
-            :value="value"
-            :sync="sync"
-            :color="colors"
-            :width="32"
-            :height="16"
-            :margin="2"
-            :disabled="disabled"
-            v-on="$listeners"
-        />
-    </span>
+        <input role="switch"
+               type="checkbox"
+               class="slider"
+               :class="[props.styleType]"
+               :disabled="props.disabled"
+               :checked="props.value"
+               @change="handleChangeToggle"
+        >
+        <span v-if="props.label !== '' && props.label !== undefined"
+              class="label"
+        >
+            {{ props.label }}
+        </span>
+    </label>
 </template>
 
-<script lang="ts">
-import {
-    defineComponent, computed, reactive, toRefs,
-} from 'vue';
-
-import { ToggleButton } from 'vue-js-toggle-button';
-
+<script setup lang="ts">
 import { TOGGLE_BUTTON_THEME } from '@/inputs/buttons/toggle-button/config';
-import type { ToggleButtonProps } from '@/inputs/buttons/toggle-button/type';
-
-import color from '@/styles/colors.cjs';
-
-/**
- * Used library: vue-js-toggle-button
- * https://www.npmjs.com/package/vue-js-toggle-button
- */
-
-export default defineComponent<ToggleButtonProps>({
-    name: 'PToggleButton',
-    components: {
-        ToggleButton,
-    },
-    props: {
-        sync: {
-            type: Boolean,
-            default: false,
-        },
-        value: {
-            type: Boolean,
-            default: false,
-        },
-        styleType: {
-            type: String,
-            default: TOGGLE_BUTTON_THEME.secondary,
-            validator(theme: TOGGLE_BUTTON_THEME) {
-                return Object.values(TOGGLE_BUTTON_THEME).includes(theme);
-            },
-        },
-        disabled: {
-            type: Boolean,
-            default: false,
-        },
-    },
 
-    setup(props: ToggleButtonProps) {
-        const state = reactive({
-            colors: computed(() => {
-                if (props.styleType === 'secondary') return { checked: color.blue[600], unchecked: color.gray[300] };
-                if (props.styleType === 'peacock500') return { checked: color.peacock[400], unchecked: color.gray[300] };
-                return { checked: color.blue[600], unchecked: color.gray[300] };
-            }),
-        });
-        return {
-            ...toRefs(state),
-        };
-    },
+interface ToggleButtonProps {
+    value: boolean,
+    label?: string,
+    styleType?: TOGGLE_BUTTON_THEME,
+    disabled?: boolean,
+}
+const props = withDefaults(defineProps<ToggleButtonProps>(), {
+    value: false,
+    label: '',
+    styleType: TOGGLE_BUTTON_THEME.secondary,
+    disabled: false,
 });
+const emit = defineEmits<{(e: 'change'): void;}>();
+const handleChangeToggle = () => {
+    emit('change');
+};
```

Comment:
suggestion: I think it would be much better if the changed value is given as an event parameter.

```suggestion
const emit = defineEmits<{(e: 'change', value: boolean): void;}>();
const handleChangeToggle = () => {
    emit('change', !props.value);
};
```

Reply:
suggestion: I think, the event name 'change' limits DX. 
Which means, if the event name is 'update:value', binding by v-model can be possible.
What about to change the event name or emit two events?